src/cli/hook-utils/manifest.js

```diff
@@ -0,0 +1,92 @@
+const fs = require('fs');
+const path = require('path');
+
+// helper array merge function
+function unionMerge(array1, array2) {
+  return [...new Set(array1.concat(array2))];
+}
+
+// look for manifest.json in the current working directory
+function readManifestJSONFile(cwd, filename, options = {}) {
+  //affordance for testing
+  const fsModule = (options.mockfs !== undefined) ? options.mockfs : fs;
```

Comment:
Here I am accepting a mock fs module as options. The scenario where this will be employed is in testing scenarios where I can pass the function a mock file system module, like what is provided by `memfs`. See the hook utilities manifest.spec.js below.

I'm curious to know if there are other ways to approach this that won't involve inserting testing related stuff into production code.

Reply:
Oof it's been a while since I've worked in Node, but agreed we should try to avoid this. It looks like this repo uses `mocha`, `chai` & `sinon`, so I think you should be able to use `sinon` to use a stub? [Something like this](https://gist.github.com/devarajchidambaram/bf05d68116bc6261add765e162deaac2).

Reply:
Agreed with @shapirone, while this module currently uses [Dependency Injection](https://en.wikipedia.org/wiki/Dependency_injection) - a fancy way of saying "passing parameters to a function" , which is a perfectly valid mocking approach and very useful in testing (we use it in the slack-cli because golang is a compiled language!) - the typical node.js way (or, really, any dynamic language) is to '[monkey patch](https://en.wikipedia.org/wiki/Monkey_patch)' a method at runtime. Essentially, dynamically overwriting a dependency's implementation with your own for e.g. testing purposes. This lets you direct execution of code during a test in a particular way by shaping how code from dependencies behave.

One example of this is in our [node-slack-sdk's oauth package's file installation store tests](https://github.com/slackapi/node-slack-sdk/blob/main/packages/oauth/src/installation-stores/file-store.spec.js#L79-L95). ["Stubs" - like test dummy stunt doubles for particular dependency methods - are created and default behaviour defined before each test run (and cleaned up/removed after each test run)](https://github.com/slackapi/node-slack-sdk/blob/main/packages/oauth/src/installation-stores/file-store.spec.js#L75-L95). Specific tests can also further refine the behaviour of stubs to adjust the execution path through the code to address specific use cases - like [having dependencies raise errors in order to exercise failure modes](https://github.com/slackapi/node-slack-sdk/blob/main/packages/oauth/src/installation-stores/file-store.spec.js#L108-L109).

We already use sinon in this repo for stubbing, mocking and spying, so I would stick to what we do already to keep the mocking method consistent.